### ISSUE TITLE
Latest code, there is an issue with DPO training reading data

### ISSUE THREAD

**Issue by vip-china** [axolotl-ai-cloud/axolotl#1256]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Latest code, there is an issue with DPO training reading data
![86e6a63cbc9797f92d77c4948d003de](https://github.com/OpenAccess-AI-Collective/axolotl/assets/119389127/61b2c3cd-ac44-4e1f-9e13-244f2725c509)

### Current behaviour

base_model: /data1/ljf2/data/dolphin-2.6-mistral-7b
model_type: MistralForCausalLM
tokenizer_type: LlamaTokenizer
is_mistral_derived_model: true

load_in_8bit: false
load_in_4bit: true
strict: false

rl: true
datasets:
  - path: /data1/ljf2/data/ultrafeedback-binarized-preferences-cleaned
    split: train
    type: ultra_apply_chatml

dataset_prepared_path: /data1/ljf2/last_data_out
val_set_size: 0.01
output_dir: /data1/ljf2/data/dpo_out

remove_unused_columns: False
adapter: qlora
lora_model_dir:

sequence_len: 16384
pad_to_sequence_len: true

lora_r: 64
lora_alpha: 32
lora_dropout: 0.05
lora_target_linear: true
lora_fan_in_fan_out:
lora_target_modules:
  - gate_proj
  - down_proj
  - up_proj
  - q_proj
  - v_proj
  - k_proj
  - o_proj
gradient_accumulation_steps: 1
micro_batch_size: 1
num_epochs: 1
optimizer: paged_adamw_8bit
lr_scheduler: cosine
learning_rate: 2e-5

train_on_inputs: false
group_by_length: false
bf16: true
fp16: false
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true

evals_per_epoch: 1
eval_table_size:
eval_table_max_new_tokens: 128

warmup_steps: 10
save_steps: 500
save_total_limit: 3
debug:
deepspeed: /data1/ljf2/data/zero3_bf16.json
weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
save_safetensors: true

### Steps to reproduce

As mentioned above

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by filippo82** [comment]
You need to modify the `type` for your dataset. It should now be `chatml.ultra`: https://github.com/OpenAccess-AI-Collective/axolotl/blob/main/docs/rlhf.md

**Comment by vip-china** [comment]
Okay, thank you very much,
